droidlet/parallel.py

```diff
@@ -0,0 +1,117 @@
+import traceback
+import queue
+from typing import Callable, List
+import cloudpickle
+# you're wondering wtf? why is numpy needed in this file?
+# it's a workaround for https://github.com/pytorch/pytorch/issues/37377
+import numpy 
+from torch import multiprocessing as mp
+from threading import Thread
+
+multiprocessing = mp.get_context("spawn")
+
+class Process(multiprocessing.Process):
+    """
+    Class which returns child Exceptions to Parent.
+    https://stackoverflow.com/a/33599967/4992248
+    """
+
+    def __init__(self, *args, **kwargs):
+        multiprocessing.Process.__init__(self, *args, **kwargs)
+        self._parent_conn, self._child_conn = multiprocessing.Pipe()
+        self._exception = None
+
+    def run(self):
+        try:
+            multiprocessing.Process.run(self)
+            self._child_conn.send(None)
+        except Exception as e:
+            tb = traceback.format_exc()
+            self._child_conn.send((e, tb))
+            raise e
+
+    @property
+    def exception(self):
+        if self._parent_conn.poll():
+            self._exception = self._parent_conn.recv()
+        return self._exception
+
+def _runner(_init_fn, init_args, _process_fn, shutdown_event, input_queue, output_queue):
+    init_fn = cloudpickle.loads(_init_fn)
+    process_fn = cloudpickle.loads(_process_fn)
+    initial_state = init_fn(*init_args)
+
+    while not shutdown_event.is_set():
+        try:
+            process_args = input_queue.get(block=True, timeout=0.1)
+            process_args_aug = (initial_state, *process_args)
+            print(process_args)
```

Comment:
replace `print`s with logging

Reply:
the `print` was removed in a later commit. it was kept in this commit by mistake.